src/clib/pio_darray.c

```diff
@@ -1525,8 +1513,43 @@ int PIOc_write_darray(int ncid, int varid, int ioid, PIO_Offset arraylen, void *
     vdesc = &(file->varlist[varid]);
     LOG((2, "vdesc record %d nreqs %d", vdesc->record, vdesc->nreqs));
 
-    ios->io_fstats->wb += iodesc->mpitype_size * iodesc->llen;
-    file->io_fstats->wb += iodesc->mpitype_size * iodesc->llen;
+    /* Run these on all tasks if async is not in use, but only on
+     * non-IO tasks if async is in use. */
+    if (!ios->async || !ios->ioproc)
+    {
+        spio_ltimer_stop(ios->io_fstats->wr_timer_name);
+        spio_ltimer_stop(ios->io_fstats->tot_timer_name);
+        spio_ltimer_stop(file->io_fstats->wr_timer_name);
+        spio_ltimer_stop(file->io_fstats->tot_timer_name);
+
+        /* Find out PIO data type of var. */
+        if ((ierr = PIOc_inq_vartype(ncid, varid, &vdesc->pio_type)))
```

Comment:
Can you also add logic to eliminate the call to inquire variable type if its already cached in vdesc?

Reply:
OK, I will add that logic, thanks.